hoomd/md/EvaluatorBondTether.h

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2009-2021 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: joaander
+#ifndef __BOND_EVALUATOR_TETHER_H__
+#define __BOND_EVALUATOR_TETHER_H__
+
+#ifndef __HIPCC__
+#include <string>
+#endif
+
+#include "hoomd/HOOMDMath.h"
+
+/*! \file EvaluatorBondTether.h
+    \brief Defines the bond evaluator class for Tethering potentials
+*/
+
+// need to declare these class methods with __device__ qualifiers when building in nvcc
+// DEVICE is __host__ __device__ when included in nvcc and blank when included into the host
+// compiler
+#ifdef __HIPCC__
+#define DEVICE __device__
+#else
+#define DEVICE
+#endif
+
+struct tether_params
+    {
+    Scalar k_b;
+    Scalar l_min;
+    Scalar l_c1;
+    Scalar l_c0;
+    Scalar l_max;
+
+#ifndef __HIPCC__
+    tether_params()
+        {
+        k_b = 10;
+        l_min = 0.9;
+        l_c1 = 1.2;
+        l_c0 = 1.8;
+        l_max = 2.1;
+        }
+
+    tether_params(Scalar k_b, Scalar l_min, Scalar l_c1, Scalar l_c0, Scalar l_max)
+        : k_b(k_b), l_min(l_min), l_c1(l_c1), l_c0(l_c0), l_max(l_max)
+        {
+        }
+
+    tether_params(pybind11::dict v)
+        {
+        k_b = v["k_b"].cast<Scalar>();
+        l_min = v["l_min"].cast<Scalar>();
+        l_c1 = v["l_c1"].cast<Scalar>();
+        l_c0 = v["l_c0"].cast<Scalar>();
+        l_max = v["l_max"].cast<Scalar>();
+
+        if (l_min < l_c1 && l_min < l_c0 && l_min < l_max && l_c1 < l_c0 && l_c1 < l_max
+            && l_c0 < l_max)
+            { /* deliberately empty to pass*/
+            }
+        else
+            {
+            throw std::invalid_argument("The input parameters should follow l_min < l_c1 < l_c0"
+                                        " < l_max");
+            }
```

Comment:
```suggestion
        if (!(l_min < l_c1 && l_min < l_c0 && l_min < l_max && l_c1 < l_c0 && l_c1 < l_max
            && l_c0 < l_max))
            {
            throw std::invalid_argument("The input parameters should follow l_min < l_c1 < l_c0"
                                        " < l_max");
            }
```
This is more readable.

Reply:
Oh yeah much readable! I'll commit suggestion right away.